model_helper.py

```diff
@@ -129,6 +129,14 @@ def compute_loss_sigmoid(logits, targets, final_sequence_length, target_sequence
 
     return loss
 
+def compute_log_probs_loss(outputs):
+    nfeatures = outputs.shape[-1] // 2
+    log_prob_ones = outputs[..., :nfeatures]
```

Comment:
I think this should not be called `log_probs`, because it is not a normalized logit. Proper `Log(P)` is never positive.

Reply:
It is named so to empathize the difference between a regular logit and this logit which is enforced to be normalized log probability by regularization term.

Reply:
Ok, got it. I think some short comment is necessary hear. Otherwise at the first glance a reader would become puzzled why we add `relu(log(P))` to the loss. Just stating that this is a pseudoprobability and that this would eventually converge to a probability due to loss terms would be good.